contracts/validator-manager/MessageSpec.md

```diff
@@ -1,185 +1,155 @@
+- `disableOwner` is the only P-Chain owner allowed to disable the validator using `DisableL1ValidatorTx`, specified below.
+
+The following is the serialization of a `PChainOwner`:
+
+|       Field |         Type |                             Size |
+| ----------: | -----------: | -------------------------------: |
+| `threshold` |     `uint32` |                          4 bytes |
+| `addresses` | `[][20]byte` | 4 + len(`addresses`) \* 20 bytes |
+|             |              | 8 + len(`addresses`) \* 20 bytes |
+
+- `threshold` is the number of `addresses` that must provide a signature for the `PChainOwner` to authorize an action.
+- Validation criteria:
+  - If `threshold` is `0`, `addresses` must be empty
+  - `threshold` <= len(`addresses`)
+  - Entries of `addresses` must be unique and sorted in ascending order
+
+### `L1ValidatorRegistrationMessage`
+
+Description: Confirms an L1 validator's registration validity on the P-Chain
+
+Signed by: P-Chain
+Consumed by: Validator Manager contract
 
 Specification:
 
-```
-SubnetValidatorWeightMessage
-+--------------+----------+----------+
-|      codecID :   uint16 |  2 bytes |
-+--------------+----------+----------+
-|       typeID :   uint32 |  4 bytes |
-+--------------+----------+----------+
-| validationID : [32]byte | 32 bytes |
-+--------------+----------+----------+
-|        nonce :   uint64 |  8 bytes |
-+--------------+----------+----------+
-|       weight :   uint64 |  8 bytes |
-+--------------+----------+----------+
-                          | 54 bytes |
-                          +----------+
-```
+|          Field |       Type |     Size |
+| -------------: | ---------: | -------: |
+|      `codecID` |   `uint16` |  2 bytes |
+|       `typeID` |   `uint32` |  4 bytes |
+| `validationID` | `[32]byte` | 32 bytes |
+|   `registered` |     `bool` |   1 byte |
+|                |            | 39 bytes |
+
+- `codecID` is the codec version used to serialize the payload, and is hardcoded to `0x0000`
+- `typeID` is the payload type identifier and is `0x00000002` for this message
+- `validationID` identifies the validator for the message
+- `registered` is a boolean representing the status of the `validationID`. If true, the `validationID` corresponds to a validator in the current validator set. If false, the `validationID` does not correspond to a validator in the current validator set, and never will in the future.
 
 ### `ValidationUptimeMessage`
 
-- Description: Provides a Validator's uptime for calculating staking rewards
+Description: Provides a Validator's uptime for calculating staking rewards
+
+Signed by: L1
+Consumed by: Validator Manager Contract
+
+Specification:
+
+|          Field |       Type |     Size |
+| -------------: | ---------: | -------: |
+|      `codecID` |   `uint16` |  2 bytes |
+|       `typeID` |   `uint32` |  4 bytes |
+| `validationID` | `[32]byte` | 32 bytes |
+|       `uptime` |   `uint64` |   8 byte |
+|                |            | 46 bytes |
```

Comment:
Where is this message spec'd?

Reply:
Defines in subnet-evm here: https://github.com/ava-labs/subnet-evm/blob/75785fac093e02c4bb893f975d59fd3f10e7b298/warp/messages/validator_uptime.go#L14